devel/spicy/reference.rst

```diff
@@ -342,9 +342,7 @@ Event Definitions
 To define a Zeek event that you want the Spicy analyzer to trigger, you
 add lines of the form::
 
-    on HOOK_ID -> event EVENT_NAME(ARG_1, ..., ARG_N);
-
-    on HOOK_ID if COND -> event EVENT_NAME(ARG_1, ..., ARG_N);
+    on HOOK_ID [if ( COND )] -> event EVENT_NAME(ARG_1, ..., ARG_N) [&priority=N];
```

Comment:
So, I have a couple points:

1) I don't think it's good to combine "generic" syntax and a "literal example" - those examples still wouldn't compile even if the `ARG_1` etc. are valid because of the `...`, so in a sense that's still not a literal example. It's a weird in between that feels more likely to be confusing, to me.
2) The later example in exporting types doesn't separate out `[...]` and literal syntaxes, but also wouldn't make much sense to because it's one very obvious addition.

I pushed a fixup commit as a bit of a middle ground, kept separate in order to easily drop it if necessary. That solves 1 by modifying the examples so they'd technically be valid if everything was in scope. Then it also fixes 2 by just getting rid of the brackets syntax for exporting types. It's a bit more verbose that way.

I'm fine to change that a bit more if it's not good. I'm not quite happy with anything I've come up with, this is the best I could do :)

Reply:
I think this is pretty good, thanks! There is always a tension between providing approachable and adaptable examples while also being technically correct, and this seems to strike a good balance.